src/main/java/com/epam/lab/paymentsystem/service/impl/OperationServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.lab.paymentsystem.service.impl;
+
+import com.epam.lab.paymentsystem.entities.Account;
+import com.epam.lab.paymentsystem.entities.Card;
+import com.epam.lab.paymentsystem.entities.Operation;
+import com.epam.lab.paymentsystem.repository.OperationRepository;
+import com.epam.lab.paymentsystem.service.OperationService;
+import com.epam.lab.paymentsystem.service.UserService;
+import java.util.List;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * This class provides implementation of the OperationService, provides methods to manipulate
+ * the data using repositories methods. The class is designed to implement business logic.
+ */
+@Service
+public class OperationServiceImpl implements OperationService {
+  private static final Logger LOGGER = LogManager.getLogger(AccountServiceImpl.class);
+
+  @Autowired
+  private OperationRepository operationRepository;
+
+  @Autowired
+  private AccountServiceImpl accountService;
+
+  @Autowired
+  private CardServiceImpl cardService;
+
+  @Autowired
+  private UserService userService;
+
+  /**
+   * Make payment operation.
+   *
+   * @param operation operation.
+   */
+  @Override
+  public void makePayment(Operation operation) {
+
+    Card sourceCard = cardService.getCardById(operation.getSourceCard().getId());
+    Card destinationCard = cardService.getCardById(operation.getTargetCard().getId());
+    Account srcAccount = sourceCard.getAccount();
+    Account dstAccount = destinationCard.getAccount();
+    LOGGER.info("Payment operation is successful");
```

Comment:
I think payment operation is successful after transaction, isn't it?

Reply:
yes, of course